numpyro/infer/mcmc.py

```diff
@@ -763,6 +761,19 @@ def sample_kernel(sa_state, model_args=(), model_kwargs=None):
         if potential_fn_gen:
             pe_fn = potential_fn_gen(*model_args, **model_kwargs)
         zs, pes, loc, scale = sa_state.adapt_state
+        # we recompute loc/scale after each iteration to avoid precision loss
```

Comment:
what's the resulting change in the computational complexity?

Reply:
It is still O(Nxd^2) where `N` is `adapt_state_size` (size of each adaptive state) and `d` is the latent dimension. The computation here is O(Nd^2) because computing covariance matrix took O(Nxd^2) while computing cholesky term is O(d^3). In a later step, we compute `log_prob` of mvn distribution with `batch_shape = (N + 1,)`, which has complexity `O(Nd^2)`. I haven't looked into details but I think `_get_proposal_loc_and_scale` also took O(Nxd^2) to compute the scale of each combination of proposals. Anyway, I believe recomputing here does not add more complexity to the overall computation.